src/resolvers-cassandra/Messages/queries.js

```diff
@@ -119,46 +116,47 @@ function byLocation(args, res) { // eslint-disable-line no-unused-vars
 }
 
 /**
- * @param {site: string, originalSource: string, bbox: number[], mainTerm: string, filteredEdges: string[], langCode: string, limit: number, offset: number, fromDate: string, toDate: string, sourceFilter: string[], fulltextTerm: string} args
+ * @param {externalsourceid: string, bbox: number[], conjunctivetopics: string[], limit: number, pageState: number, fromDate: string, toDate: string, pipelinekeys: string[], fulltextTerm: string} args
  * @returns {Promise.<{runTime: string, type: string, bbox: number[], features: Feature[]}>}
- */
+ **/
 function byBbox(args, res) { // eslint-disable-line no-unused-vars
   return new Promise((resolve, reject) => {
     if (!args.bbox || args.bbox.length !== 4) return reject('Invalid bbox specified');
+    if (!args.conjunctivetopics.length) return reject('Empty conjunctive topic list specified');
 
-    const { fromDate, toDate } = parseFromToDate(args.fromDate, args.toDate);
     const [north, west, south, east] = args.bbox;
 
+    let tableName = 'eventplaces';
+    let tagsParams = [
+      ...fromTopicListToConjunctionTopics(args.conjunctivetopics),
+      args.toDate,
+      north,
+      east,
+      args.fromDate,
+      south,
+      west,
+      args.pipelinekeys
```

Comment:
Shouldn't this only be included if the `args.pipelinekeys` is truthy? Otherwise we'll have the extra undefined param in here but nothing to bind it to in the query below.

Example to make it clearer:
1) Assume args.pipelinekey is null
2) The tagsParams array will be [..., null] since args.pipelinekey is null.
3) The tagsQuery will not include the `AND pipelinekey in ?` since it's guarded by the if statement.
4) Now we have N params to bind in tagsParams and N-1 placeholders (`?`) in the query so there should be an error.

To fix it, I would suggest this:

```js
const tagsParams = [
  ...fromTopicListToConjunctionTopics(args.conjunctivetopics),
  args.toDate,
  north,
  east,
  args.fromDate,
  south,
  west,
];

if (args.pipelinekey) {
  tagsParams.push(args.pipelinekey);
}
```